src/manager.rs

```diff
@@ -329,23 +373,27 @@ pub async fn manager_main(mut goose_attack: GooseAttack) -> GooseAttack {
                     // We need another worker, accept the connection.
                     else {
                         // Validate worker load test hash.
-                        match requests.get("load_test_hash") {
-                            Some(r) => {
-                                if r.load_test_hash != goose_attack.stats.hash {
+                        if let Some(requests) = gaggle_stats.requests {
```

Comment:
This case is actually invalid and a case for panic() as well, so I would either use `let requests = gaggle_stats.requests.expect('...')` OR `let requests = gaggle_stats.requests.unwrap()?;` - if we can pass the error up.

Adding more indentation does actually hide an error case.

Reply:
To explain:

```
if let Some(requests) = gaggle_stats.requests
```

is plain wrong, let's assume gaggle_stats.requests is None, then the error is silently discarded.

You want:

```
match gaggle_stats.requests.expect('Request must not be empty').get("load_test_hash") {
```

instead.

Reply:
The way the logic is written, requests won't be empty at this time, as we start by sending "fake packets". In any case, the Option<> goes away when we switch to an enum, so this will go away.

Reply:
Then just use .unwrap() - let Some(x) is for cases where the None option is a valid case and mit one that can never happen.

Reply:
We don't want the Manager to panic if the wrong number of Workers are started, instead we have the extra Worker's panic -- the Manager and correct Workers can safely continue their load test.

This area of the codebase has been reworked to support Enums, and I think it's cleaner now.